loadgen/logging.cc

```diff
@@ -518,14 +500,39 @@ bool TlsLogger::ReadBufferHasBeenConsumed() {
   return unread_swaps_ == 0;
 }
 
+void TlsLogger::TraceCounters() {
+  auto trace = MakeScopedTracer(
+      [lcfc = log_cas_fail_count_.load(std::memory_order_relaxed),
+       sbsrc = swap_buffers_slot_retry_count_.load(std::memory_order_relaxed)]
+        (AsyncLog& log) {
+    log.ScopedTrace("TlsLogger:ContentionCounters",
+                    "log_cas_fail_count", lcfc,
+                    "swap_buffers_slot_retry_count", sbsrc);
+  });
+}
+
 Logger& GlobalLogger() {
   static Logger g_logger(kLogPollPeriod, kMaxThreadsToLog);
   return g_logger;
 }
 
+// TlsLoggerWrapper moves ownership of the TlsLogger to Logger on thread exit
+// so no round-trip synchronization with the IO thread is required.
+struct TlsLoggerWrapper {
```

Comment:
@guschmue : Hoping this will reduce the potential for deadlock if the IOThread is disabled before the TlsLoggers.